src/dart/codegen.rs

```diff
@@ -1,129 +1,215 @@
+            var_ty = self.codegen_type(ty);
+        }
+        if let Some(ref expr) = field.default {
+            var_expr = Some(self.codegen_expr(expr));
+        }
+        ast::ClassMember::Fields {
+            metadata: vec![],
+            static_: false,
+            var_type: ast::VarType {
+                fcv: ast::FinalConstVar::Final,
+                ty: var_ty,
+            },
+            initializers: vec![
+                ast::NameAndInitializer {
+                    name: field.name,
+                    init: var_expr,
+                },
+            ],
         }
-        self.write_ident(&field.name);
-        self.write_str(";\n");
     }
 
-    fn codegen_field(&mut self, field: &Field) {
-        self.write_ident(&field.name);
-        self.write_str(": ");
-        self.codegen_expr(&field.value);
-        self.write_str(",\n");
+    fn codegen_field(&mut self, field: &Field) -> ast::NamedArg {
+        ast::NamedArg {
+            name: field.name,
+            expr: self.codegen_expr(&field.value),
+        }
     }
 
-    fn codegen_expr(&mut self, expr: &Expr) {
+    fn codegen_expr(&mut self, expr: &Expr) -> Node<ast::Expr> {
         match *expr {
             Expr::Instance(ref instance) => self.codegen_instance(&instance),
             Expr::Array(ref exprs) => {
-                self.write_str("[\n");
-                for expr in exprs {
-                    self.codegen_expr(expr);
-                    self.write_str(",\n");
-                }
-                self.write_str("]");
-            }
-            Expr::Dart(ref dart) => {
-                self.tokens.extend(dart);
+                let elements = exprs.iter().map(|expr| self.codegen_expr(expr)).collect();
+                Node::new(ast::Expr::List {
+                    const_: false,
+                    element_ty: None,
+                    elements,
+                })
             }
+            Expr::Dart(ref dart) => dart.clone(),
         }
+
     }
 
-    fn codegen_type(&mut self, ty: &Type) {
+    fn codegen_type(&mut self, ty: &Type) -> Node<ast::Type> {
         match *ty {
-            Type::Dart(ref dart) => {
-                self.tokens.extend(dart);
-            }
+            Type::Dart(ref dart) => dart.clone(),
         }
     }
 
-    fn codegen_instance(&mut self, instance: &Instance) {
-        self.write_str("new ");
-        self.write_ident(&instance.name);
-        self.write_str("(\n");
-        for field in &instance.fields {
-            self.codegen_field(field);
-        }
-        self.write_str(")");
+    fn codegen_instance(&mut self, instance: &Instance) -> Node<ast::Expr> {
+        let args = instance
+            .fields
+            .iter()
+            .map(|field| self.codegen_field(field))
+            .collect();
+        Node::new(ast::Expr::New {
+            const_: false,
+            ty: Node::new(ast::Type::Path(
+                ast::Qualified::simple(&instance.name.as_str()),
```

Comment:
This should manually create the `Qualified` instead of going through `.as_str()` (`instance.name` is already a `Symbol`).